File: serverless/src/userPaymentsHandler.js
import httpResponse from "./http_response.js";
import { retrievePayments } from "./userPayments.js";

export const handler = async (event) => {
  try {
    switch (event.httpMethod) {
      case "GET":
        const customerId = event.queryStringParameters.customer_id;
        return await retrievePayments(customerId);
    }
  } catch (err) {
    return httpResponse(err.statusCode, err.message);
  }
};

File: serverless-python/src/user_payments_handler.py
from error_handler import handle_client_error
from botocore.exceptions import ClientError
from user_payments import retrievePayments

def handler(event, _):
  query_params = event.get('queryStringParameters', {})
  email = query_params.get('email')
  try:
      return retrievePayments(email)
  except ClientError as err:
      handle_client_error(err)
